multisig/multisig.go

```diff
@@ -1,88 +1,376 @@
+	}
+	return txBytes, nil
 }
 
-func (*Multisig) ToFile(_ string) error {
+func (ms *Multisig) ToFile(txPath string) error {
+	if ms.Undefined() {
+		return ErrUndefinedTx
+	}
+	txBytes, err := ms.ToBytes()
+	if err != nil {
+		return err
+	}
+	txStr, err := formatting.Encode(formatting.Hex, txBytes)
+	if err != nil {
+		return fmt.Errorf("couldn't encode signed tx: %w", err)
+	}
+	f, err := os.Create(txPath)
+	if err != nil {
+		return fmt.Errorf("couldn't create file to write tx to: %w", err)
+	}
+	defer f.Close()
+	_, err = f.WriteString(txStr)
+	if err != nil {
+		return fmt.Errorf("couldn't write tx into file: %w", err)
+	}
 	return nil
 }
 
-func (*Multisig) FromFile(_ string) error {
+func (ms *Multisig) FromBytes(txBytes []byte) error {
+	var tx txs.Tx
+	if _, err := txs.Codec.Unmarshal(txBytes, &tx); err != nil {
+		return fmt.Errorf("error unmarshaling signed tx: %w", err)
+	}
+	if err := tx.Initialize(txs.Codec); err != nil {
+		return fmt.Errorf("error initializing signed tx: %w", err)
+	}
+	ms.pChainTx = &tx
 	return nil
 }
 
-func (*Multisig) Sign(_ *primary.Wallet) error {
-	return nil
+func (ms *Multisig) FromFile(txPath string) error {
+	txEncodedBytes, err := os.ReadFile(txPath)
+	if err != nil {
+		return err
+	}
+	txBytes, err := formatting.Decode(formatting.Hex, string(txEncodedBytes))
+	if err != nil {
+		return fmt.Errorf("couldn't decode signed tx: %w", err)
+	}
+	return ms.FromBytes(txBytes)
 }
 
-func (*Multisig) Commit() error {
-	return nil
+func (ms *Multisig) IsReadyToCommit() (bool, error) {
+	if ms.Undefined() {
+		return false, ErrUndefinedTx
+	}
+	_, remainingSigners, err := ms.GetRemainingAuthSigners()
+	if err != nil {
+		return false, err
+	}
+	return len(remainingSigners) == 0, nil
 }
 
-func (*Multisig) IsReadyToCommit() error {
-	return nil
+// get subnet auth addresses that did not yet signed a given tx
+//   - get the string slice of auth signers for the tx (GetAuthSigners)
+//   - verifies that all creds in tx.Creds, except the last one, are fully signed
+//     (a cred is fully signed if all the signatures in cred.Sigs are non-empty)
+//   - computes remaining signers by iterating the last cred in tx.Creds, associated to subnet auth signing
+//   - for each sig in cred.Sig: if sig is empty, then add the associated auth signer address (obtained from
+//     authSigners by using the index) to the remaining signers list
+//
+// if the tx is fully signed, returns empty slice
+func (ms *Multisig) GetRemainingAuthSigners() ([]ids.ShortID, []ids.ShortID, error) {
+	if ms.Undefined() {
+		return nil, nil, ErrUndefinedTx
+	}
+	authSigners, err := ms.GetAuthSigners()
+	if err != nil {
+		return nil, nil, err
+	}
+	emptySig := [secp256k1.SignatureLen]byte{}
+	// we should have at least 1 cred for output owners and 1 cred for subnet auth
+	if len(ms.pChainTx.Creds) < 2 {
+		return nil, nil, fmt.Errorf("expected tx.Creds of len 2, got %d", len(ms.pChainTx.Creds))
```

Comment:
nit: fix error. current wording `expected tx.Creds of len 2, got %d` doesn't give enough info to the user what to do to fix it